cub/agent/agent_spmv_orig.cuh

```diff
@@ -458,22 +429,37 @@ struct AgentSpmv
             #pragma unroll
             for (int ITEM = 0; ITEM < ITEMS_PER_THREAD; ++ITEM)
             {
-                int     nonzero_idx             = threadIdx.x + (ITEM * BLOCK_THREADS);
-                nonzero_idx                     = CUB_MIN(nonzero_idx, tile_num_nonzeros - 1);
-
-                OffsetT column_idx              = wd_column_indices[tile_start_coord.y + nonzero_idx];
-                ValueT  value                   = wd_values[tile_start_coord.y + nonzero_idx];
-
-                ValueT  vector_value            = wd_vector_x[column_idx];
+                int nonzero_idx = threadIdx.x + (ITEM * BLOCK_THREADS);
 
-                ValueT  nonzero                 = value * vector_value;
+                // clang-format off
+                NV_IF_TARGET(NV_PROVIDES_SM_52,
+                (
+                    auto a = wd_values + tile_start_coord.y + nonzero_idx;
```

Comment:
Is there a reason we need those computations outside of the if?

and obligatory hiss at one letter variable names